### PR TITLE
add test files for log.go and rule.go

### PR DESCRIPTION
**Pull request by sunzhuohang**
### What problem does this PR solve?
Add test files for log.go and rule.go. Improve the coverage of server/api. #2287

### CODE REVIEW

**Comment by disksing** [comment]
Thanks. Please add lisence at the begining of each file.

File: server/api/rule_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	postData, err = json.Marshal(rule3)
+	c.Assert(err, IsNil)
+	err = postJSON(testDialClient, s.urlPrefix+"/rule", postData)
+	c.Assert(err, IsNil)
+
+	//Get
+	var resp placement.Rule
+	err = readJSON(testDialClient, s.urlPrefix+"/rule/pd/default", &resp)
+	c.Assert(err, IsNil)
+	c.Assert(resp.Count, Equals, 3)
```
Comment (nolouch) [inline server/api/rule_test.go:87]: Can you check if the response content is as expected？

File: server/api/log_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (rleungx) [inline server/api/log_test.go:1]:
```suggestion
// Copyright 2020 PingCAP, Inc.
```

File: server/api/log_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	. "github.com/pingcap/check"
```
Comment (rleungx) [inline server/api/log_test.go:19]: It's better to add an empty line.

File: server/api/rule_test.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (rleungx) [inline server/api/rule_test.go:1]:
```suggestion
// Copyright 2020 PingCAP, Inc.
```

File: server/api/rule_test.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
```
Comment (rleungx) [inline server/api/rule_test.go:18]: ditto and move `"net/http"` to line 19

File: server/api/log_test.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	mustBootstrapCluster(c, s.svr)
+}
+
+func (s *testLogSuite) TearDownSuite(c *C) {
+	s.cleanup()
+}
+
+func (s *testLogSuite) TestSetLogLevel(c *C) {
+	level := "error"
+	if log.GetLevel().String() == level {
```
Comment (rleungx) [inline server/api/log_test.go:48]: I think in this test, it won't execute line 49?

Comment (sunzhuohang) [reply server/api/log_test.go:48]: If the original level is error. It‘s uncertain whether setloglevel is executed correctly.

Comment (sunzhuohang) [reply server/api/log_test.go:48]: > I think in this test, it won't execute line 49?
If the original level is error. It‘s uncertain whether setloglevel is executed correctly.

Comment (rleungx) [reply server/api/log_test.go:48]: I mean in this test, `log.GetLevel().String()` will be `info`, this if statement will always be false?

**Pull request merged**